compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp

```diff
@@ -89,6 +91,17 @@ enumerateMatmulTileArm64(EncodingUser user, TypeRange elementTypes,
   Type rhs = elementTypes[1];
   Type out = elementTypes[2];
 
+  uint32_t ukType = findMmt4dUkernelType(lhs, rhs, out);
+  // TODO(#15784): If ukernel is enabled and the element types are not
+  // supported, currently CPULowerToUKernels will simply fail and lower into
+  // scalar loops. Don't set encoding for now. Otherwise codegen should still
+  // be able to generate code for ukernel unsupported types.
+  //
+  // This should be removed once CPULowerToUKernels can fallback to codegen.
+  if (hasUkernel(target) && ukType == IREE_UK_FLAG_MMT4D_TYPE_NONE) {
+    return {};
+  }
```

Comment:
I think we can remove the logic because we have a reasonable fallback for ukernel path: https://github.com/openxla/iree/commit/33a331be781025e7012cae4e4e16d476edd5cf5f

IMO, data-tiling is always unconditionally good to apply to matmuls. It is better to make it be applied to both, regardless of ukernels.

Reply:
I added the separate tile sizes for DT-only path, similar to x86_64